internal/providers/terraform/google/container_cluster.go

```diff
@@ -22,87 +20,115 @@ func GetContainerClusterRegistryItem() *schema.RegistryItem {
 	}
 }
 
-func NewContainerCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
-	description := "Regional Kubernetes Clusters"
+func newContainerCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
 	region := d.Get("location").String()
-	if isZone(region) {
+	isZone := isZone(region)
+
+	if isZone {
 		region = zoneToRegion(region)
-		description = "Zonal Kubernetes Clusters"
 	}
 
-	subResources := make([]*schema.Resource, 0)
+	autopilotEnabled := d.Get("enable_autopilot").Bool()
+
+	var defaultNodePool *google.ContainerNodePoolConfig
+	nodePools := make([]*google.ContainerNodePoolConfig, 0)
 
-	if !d.Get("remove_default_node_pool").Bool() {
+	if !d.Get("remove_default_node_pool").Bool() && !autopilotEnabled {
 		zones := int64(zoneCount(d.RawValues, ""))
 
 		countPerZone := int64(3)
 
-		if d.Get("initial_node_count").Type != gjson.Null {
+		if !d.IsEmpty("initial_node_count") {
```

Comment:
Spent some time debugging an issue with 0 node count. The code used `d.Get("initial_node_count").Exists()`. HCL showed correct result, but TF plan returned this as true and set `countPerZone` to 0. Switching to `IsEmpty()` fixed it.